src/Components/Components/src/Routing/Router.cs

```diff
@@ -23,8 +23,8 @@ namespace Microsoft.AspNetCore.Components.Routing
     public class Router : IComponent, IHandleAfterRender, IDisposable
     {
         static readonly char[] _queryOrHashStartChar = new[] { '?', '#' };
-        static readonly ReadOnlyDictionary<string, object> _emptyParametersDictionary
-            = new ReadOnlyDictionary<string, object>(new Dictionary<string, object>());
+        static readonly IReadOnlyDictionary<string, object> _emptyParametersDictionary
```

Comment:
Why isn't this the concrete type to begin with?

Reply:
I had the same question. Probably because `RouteContext.Parameters` is `IReadOnlyDictionary<string, object>`.

Reply:
It's used as a parameter elsewhere which is of type IReadOnlyDictionary

Reply:
Could you add a comment that says not to change this to ReadOnlyDictionary? It's very tempting for someone to accidentally undo this change in the name of correctness of code.

Reply:
```suggestion
        // Dictionary is intentionally used instead of ReadOnlyDictionary to reduce Blazor size
        static readonly IReadOnlyDictionary<string, object> _emptyParametersDictionary
```

Reply:
> Why isn't this the concrete type to begin with?

My reasoning was so no one could accidently add anything to this dictionary, since the field is `IReadOnlyDictionary`.